plugins/c/hu.bme.mit.gamma.xsts.codegeneration.c/src/hu/bme/mit/gamma/xsts/codegeneration/c/serializer/HavocSerializer.xtend

```diff
@@ -0,0 +1,121 @@
+/********************************************************************************
+ * Copyright (c) 2018-2023 Contributors to the Gamma project
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ ********************************************************************************/
+package hu.bme.mit.gamma.xsts.codegeneration.c.serializer
+
+import hu.bme.mit.gamma.expression.model.BooleanTypeDefinition
+import hu.bme.mit.gamma.expression.model.DecimalTypeDefinition
+import hu.bme.mit.gamma.expression.model.DirectReferenceExpression
+import hu.bme.mit.gamma.expression.model.EnumerationTypeDefinition
+import hu.bme.mit.gamma.expression.model.Expression
+import hu.bme.mit.gamma.expression.model.IntegerTypeDefinition
+import hu.bme.mit.gamma.expression.model.RationalTypeDefinition
+import hu.bme.mit.gamma.expression.model.Type
+import java.util.Random
+
+/**
+ * Serializes different types of definitions and expressions into their string representation.
+ * Supports boolean, integer, decimal, rational, and enumeration types.
+ * Also supports direct reference expressions.
+ */
+class HavocSerializer {
+	
+	Random random = new Random;
+	
+	/**
+     * Throws an exception for unsupported types.
+     *
+     * @param type the type to be serialized
+     * @param name the name of the type to be serialized
+     * @return an exception message for unsupported types
+     * @throws IllegalArgumentException if the type is not supported
+     */
+	def dispatch String serialize(Type type, String name) {
+		throw new IllegalArgumentException("Not supported type: " + type);
+	}
+	
+	/**
+     * Serializes boolean types into their string representation.
+     *
+     * @param type the boolean type to be serialized
+     * @param name the name of the boolean type to be serialized
+     * @return the string representation of the serialized boolean type
+     */
+	def dispatch String serialize(BooleanTypeDefinition type, String name) {
+		return random.nextBoolean().toString();
```

Comment:
In case we decide to keep supporting the `HavocAction`, I propose a solution that introduces yet another source + header pair that contain pre-generated functions for each data type within the model, returning runtime generated random values (int, bool, literals, etc..). The approach trades some overhead due to potentially unnecessary functions for simplicity that leaves less room for error. I believe the required import is supported by the ISO C Standard, therefore a platform-independent solution is possible without further modification to the `platforms` package.

Reply:
Okay, let us go with this solution. However, before we generate the import, we should check whether the merged action in the XSTS contains any havoc actions (see `GammaEcoreUtil` class and `contains...` methods) so that we do not include this part unnecessarily.